### ISSUE TITLE
qsturng.py Inaccuracy

### ISSUE THREAD

**Issue by DominicChm** [raphaelvallat/pingouin#155]
Hello,

I was doing some testing recently, and noticed that the `qstrung.py` implementation in Pingouin seems to give a large error when calling pstrung. The implementation, which seems almost identical to Statsmodels', gives errors on the order of 10^-2 when compared with manual integral evaluation, R, and Statmodels.

Here's a quick test script to demonstrate what I mean: 
```python
from pingouin.external.qsturng import psturng as ping_q
from statsmodels.stats.libqsturng import psturng as stats_q

import numpy as np
import itertools

q_i = np.arange(1, 4)
k_i = range(2, 6)
v_i = range(2, 120, 10)

tot_err = 0
total = 0

print("Calculating average error")
for q, k, v in itertools.product(q_i, k_i, v_i):
	tot_err += abs(stats_q(q, k, v) - ping_q(q, k, v))
	total += 1

print(f"Average stock error: {tot_err / total}")
```
```
> Calculating average error
> Average stock error: [0.015114]
```

It looks like the problem is that somewhere along the way, some `min()` statements got switched out for some `max()` ones.

Pingouin:
```python
    return math.sqrt(2) * -y * \
           scipy.stats.t.isf((1. + p) / 2., max(v, 1e38))
```

Statsmodels:
```python
    v = min(v, 1e38)
    return math.sqrt(2) * -y * scipy.stats.t.isf((1. + p) / 2., v)
```

The mismatches are on **lines 580, 582, 592, 725**
Simply replacing all the `max()` statements with `min()` statements seems to fix the problem. 
```
> Calculating patched error
> Average patched error: [0.]
```

Let me know if you'd like me to submit a PR. 

Thanks!

**Comment by raphaelvallat** [comment]
Hi @DominicChm,

Thanks for spotting this issue. The [qsturng.py](https://github.com/raphaelvallat/pingouin/blob/master/pingouin/external/qsturng.py) file is basically a copy-paste from statsmodels a few years ago when I first started working on Pingouin. My guess is that they've updated the file since. We could update the file, or even better, I think we could simply remove it (and the entire external/ folder) and instead directly call statsmodels (which is now a dependency of Pingouin anyway). Does that sound like a good approach?

Second, I would be curious to see how much of a difference this issue makes in the calculation of the p-values in the 
 pingouin.pairwise_tukey and pingouin.pairwise_gameshowell functions (i.e. the only times where the psturng is actually used in Pingouin):

https://github.com/raphaelvallat/pingouin/blob/58df89197905c7382ee721e57c8abf9fa748e7df/pingouin/pairwise.py#L711

For these two functions, Pingouin was tested against R and JASP, with results that are very similar, although the p-values did differ a little bit, which I attributed to differences in the algorithm used to calculate the studentized range approximation

https://github.com/raphaelvallat/pingouin/blob/58df89197905c7382ee721e57c8abf9fa748e7df/pingouin/tests/test_pairwise.py#L373

So I'd be very curious to see if we can now replicate the exact p-values of R or JASP.

Thanks again!
Raphael
